src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -462,10 +468,23 @@ private void CheckModifiers(Location location, bool hasBody, bool isAutoProperty
             {
                 diagnostics.Add(AccessCheck.GetProtectedMemberInSealedTypeError(ContainingType), location, this);
             }
-            else if (IsStatic && IsDeclaredReadOnly && !_property.HasReadOnlyModifier)
+            else if (LocalDeclaredReadOnly)
             {
-                // The modifier '{0}' is not valid for this item
-                diagnostics.Add(ErrorCode.ERR_BadMemberFlag, location, SyntaxFacts.GetText(SyntaxKind.ReadOnlyKeyword));
+                if (_property.HasReadOnlyModifier)
+                {
+                    // Cannot specify 'readonly' modifiers on both property or indexer '{0}' and its accessors.
+                    diagnostics.Add(ErrorCode.ERR_InvalidPropertyReadOnlyMods, location, _property);
+                }
+                if (IsStatic)
```

Comment:
>if [](start = 16, length = 2)

Is the `if` rather than `else if` intentional? Are we testing: a `static readonly` properties with `readonly` accessor; and a `readonly` auto-property with `readonly` accessor?

Reply:
Oops, this should be `if`/`else if`/`else if`. Maybe this is an argument for flattening it out and doing repeated checks of `LocalDeclaredReadOnly`?

---
In reply to: [267975807](https://github.com/dotnet/roslyn/pull/34338#discussion_r267975807) [](ancestors = 267975807)